tests/benchmark/compute/instruction/test_system.py

```diff
@@ -723,18 +730,29 @@ def test_selfdestruct_created(
         balance=iterations if value_bearing else 0,
         storage={0: 1},
     )
-    code_tx = Transaction(
-        to=code_addr,
-        gas_limit=gas_benchmark_value,
-        sender=pre.fund_eoa(),
-    )
 
-    post = {code_addr: Account(storage={0: 42})}  # Check for successful
-    # execution.
-    state_test(
-        pre=pre,
+    max_iterations_per_tx = (tx_gas_limit - base_costs) // loop_cost
+    num_exec_txs = math.ceil(iterations / max_iterations_per_tx)
+
+    exec_txs = []
+    with TestPhaseManager.execution():
+        for _ in range(num_exec_txs):
+            exec_txs.append(
+                Transaction(
+                    to=code_addr,
+                    gas_limit=tx_gas_limit,
+                    sender=pre.fund_eoa(),
+                )
+            )
+
+    post = {
+        code_addr: Account(storage={0: len(exec_txs) + 1})
+    }  # Check for successful execution.
+    benchmark_test(
         post=post,
-        tx=code_tx,
+        blocks=[
+            Block(txs=exec_txs),
+        ],
         expected_benchmark_gas_used=iterations * loop_cost + base_costs,
```

Comment:
This does not consider the multiple transactions implementation, we need to update as:


```suggestion
        expected_benchmark_gas_used=(
            max_iterations_per_tx * loop_cost + base_costs
        )
        * num_exec_txs
```